core/src/main/scala/com/evolutiongaming/kafka/flow/PartitionFlow.scala

```diff
@@ -188,12 +188,13 @@ object PartitionFlow {
 
       clock <- Clock[F].instant
       timestamp <- timestamp updateAndGet (_.copy(clock = clock))
-      states <- cache.values
-      _ <- states.values.toList.parTraverse_ { state =>
-        state flatMap { state =>
-          state.timers.set(timestamp) *>
-            state.timers.trigger(state.flow)
-        }
+
+      keys <- cache.keys
+      states <- keys.toVector.traverseFilter(cache.get)
+
+      _ <- states.parTraverse_ { state =>
+        state.timers.set(timestamp) *>
+          state.timers.trigger(state.flow)
```

Comment:
is this a heavy operation?

Reply:
Yes, it can include sending snapshots to Kafka, Cassandra etc.